pulpcore/cli/core/task.py

```diff
@@ -1,41 +1,99 @@
+from typing import Any, Dict, List
+
 import click
 
-from pulpcore.cli.common import main, PulpContext
+from pulpcore.cli.common import DEFAULT_LIMIT, BATCH_SIZE, PulpContext
+
+
+class PulpTaskContext:
+    def __init__(self, pulp_ctx: PulpContext) -> None:
+        self.pulp_ctx: PulpContext = pulp_ctx
+
+    def list(self, limit: int, parameters: Dict[str, Any]) -> List[Any]:
+        count: int = -1
+        entities: List[Any] = []
+        payload: Dict[str, Any] = parameters.copy()
+        payload["offset"] = 0
+        payload["limit"] = BATCH_SIZE
+        while limit != 0:
+            if limit > BATCH_SIZE:
+                limit -= BATCH_SIZE
+            else:
+                payload["limit"] = limit
+                limit = 0
+            result: Dict[str, Any] = self.pulp_ctx.call("tasks_list", parameters=payload)
+            count = result["count"]
+            entities.extend(result["results"])
+            if result["next"] is None:
+                break
+            payload["offset"] += payload["limit"]
+        else:
+            click.echo(f"Not all {count} entries were shown.", err=True)
+        return entities
```

Comment:
As you can see, this code is (hopefully) generic enough to be reused by other entities.